lib/parser.js

```diff
@@ -21,15 +21,31 @@ function Html2ReactParser(options) {
     });
   };
 
-  function traverseDom(node, isValidNode, processingInstructions, index) {
+  function traverseDom(
+    node,
+    isValidNode,
+    processingInstructions,
+    preprocessingInstructions,
+    index
+  ) {
+
     if (isValidNode(node)) {
+      if (preprocessingInstructions) {
+        preprocessingInstructions.map(function (instruction) {
+          if (instruction.shouldPreprocessNode(node)) {
+            instruction.preprocessNode(node, children, index);
```

Comment:
I can't see how `children` has been defined at this point? Could you also please specify the purpose of the `children` argument to preprocessors?

Reply:
The examples don't use `children` or `index` but it can be beneficial to have that data. I cannot think of any use-case where it is necessary though but kept it there in case. Removed it now and we can add later if its needed.